src/components/Queue.vue

```diff
@@ -37,20 +46,95 @@
         </div>
 
     </div>
-
+    </transition>
 </template>
 
 <script>
 import Store from '../store.js';
+
+const RESULT_TIMER = 5000;
+
 export default {
     name: 'queue',
     data: function() {
         return {
+            queue: Store.queue,
             collapsed : true,
             show : true,
-            vouchers : Store.trader.vouchers
+            netMgr: Store.netMgr,
+            vouchers : Store.trader.vouchers,
+            spinner: false,
+            validate: false,
+            fail: false,
+            message: '',
         }
-    }
+    },
+    watch: {
+      queue: {
+          handler: function(val, oldVal) {
+              let queueState = val.sendingStatus || false;
+              if(!queueState && this.netMgr.online) {
+                  this.showValidate();
+              } else if(!queueState) {
+                  this.spinner = false;
+              }
+
+          },
+          deep: true
+      },
 
+    },
+    mounted: function() {
+        if(Store.queue.sendingStatus) {
+            this.startSpinner();
+        }
+    },
+    computed: {
+        currentlyShown: function() {
+            return (this.fail || this.validate || (this.vouchers.length >= 1 && !this.netMgr.online) || this.queue.sendingStatus);
+        }
+    },
+    methods: {
+        startSpinner: function() {
+            this.spinner = true;
+        },
+
+        showValidate: function() {
+            this.spinner = false;
+            this.validate = true;
+
+            this.message = "[xXx] Thanks! We've successfully submitted your queued vouchers.";
+            setTimeout(function() {
+                this.validate = false;
+                this.message = '';
+            }.bind(this), RESULT_TIMER);
+        },
+
+        showFail: function() {
+            this.spinner = false;
+            this.fail = true;
+
+            this.message = "[xXx] Whoops! There may be a network problem. When you have a better signal, click 'Submit queued vouchers' to retry.";
```

Comment:
@gingerbreadpup At this stage let's get the text in as approved if we can so we don't have to card another xXx sweep.